CMSgov/mdct-mcr
# create global layout (header, footer, and more)

## Description


Create global layout, including header and footer, and adds appropriate unit and integration tests. It looks like a ton of lines, but that's like 80% `yarn.lock` fun, so don't be too alarmed.

Incidentally introduces the following:
- New reusable custom components (Menu, MenuOption) and custom wrappers (RouterLink, Icon).
- Tabbed navigation focus functionality.
- New custom hooks (useBreakpoint, makeMediaQueryClasses) to replace media queries and simplify responsive styling efforts.
- Global styling baseline and patterns for styling solution.
- Theme breakpoints.
- Integration testing functionality and sets baseline.

Satisfies the following tickets: 
- [MDCT-168: MCR Global Header](https://qmacbis.atlassian.net/browse/MDCT-168)
- [MDCT-169: MCR Global Footer](https://qmacbis.atlassian.net/browse/MDCT-169)

### How to test


Prerequisite: Ensure correct `.env` (see below).

To test UI:
1. `cd` && `nvm use`
2. `./dev local`
3. Check things out and try to break it (mess with responsive breakpoints, click around, etc.). Note: All routes outside of `/` are non-functional.

Unit tests:
1. `cd services/ui-src && nvm use`
2. `yarn test`
3. All tests should be passing

Integration tests:
1. `cd` && `nvm use`
2. `yarn test`
3. When cypress window launches, click on any test suite to run.
4. All tests should be passing and there should be no accessibility errors or warnings.

### Changed Dependencies

Just a little `yarn upgrade` action for `/`, `/services/ui-src`, and `/tests/cypress`.

## Code author checklist

## Reviewer checklist (two different people)

***
***
## Environment Configuration

#### root `.env`
```
SKIP_PREFLIGHT_CHECK=true
LOCAL_LOGIN=true
MEASURE_TABLE_NAME=local-measures
MEASURE_TABLE_ARN=local_nonsense_if_unset_we_search_CF_for
measureTableName=local-measures
DYNAMODB_URL=http://localhost:8000/
API_URL=http://localhost:3030/local
S3_LOCAL_ENDPOINT=http://localhost:4569/
S3_ATTACHMENTS_BUCKET_NAME=local-uploads
URL=http://localhost/3000
```

#### services/ui-src `.env`
```
LOCAL_LOGIN=true
API_REGION=us-east-1
API_URL=http://localhost:3030/
COGNITO_REGION=us-east-1
COGNITO_IDENTITY_POOL_ID=us-east-1:61c849d8-f95c-45af-8131-15b75c525034
COGNITO_USER_POOL_ID=us-east-1_7U4ihGYZP
COGNITO_USER_POOL_CLIENT_ID=46fh6sftchq768ab0v65cpq0cf
COGNITO_USER_POOL_CLIENT_DOMAIN=master-login-46fh6sftchq768ab0v65cpq0cf.auth.us-east-1.amazoncognito.com
COGNITO_REDIRECT_SIGNIN=http://localhost:3000/
COGNITO_REDIRECT_SIGNOUT=http://localhost:3000/
S3_ATTACHMENTS_BUCKET_REGION=us-east-1
S3_ATTACHMENTS_BUCKET_NAME=uploads-master-attachments-024259748323
S3_LOCAL_ENDPOINT=http://localhost:4569/
```
